department-of-veterans-affairs/vets-api
# Retryable SubmitForm686cJob and SubmitForm674Job (Pending Full Idempotence)

## Summary
- Allow `SubmitForm686cJob` and `SubmitForm674Job` sidekiq jobs to be retryable up to 14 times
  - Log error after each attempt but only send email and attempt to recover form on final attempt
- Partial idempotence (only if there are errors)
- Move error handling to only send email and save form after retries are exhausted
  - Additionally move user object logic to top of jobs so that error handling can always have access to user object

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/65052
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/56290
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/65151


## Testing done
- Create a claim via /view-change-dependents/add-remove-form-21-686c/
- Check in /sidekiq admin console that claim will retry if there is an error (locally it will fail always)

## Screenshots
![Screenshot 2023-09-19 at 16 46 40](https://github.com/department-of-veterans-affairs/vets-api/assets/583973/5c028b3e-2bc4-4ae4-87cd-7d74e8414416)


## What areas of the site does it impact?
None are functionally impacted, this should only assist in background jobs being handled better after errors

## Acceptance criteria
